src/bun.js/bindings/webcore/JSEventEmitterCustom.cpp

```diff
@@ -0,0 +1,52 @@
+#include "config.h"
+#include "JSEventEmitter.h"
+
+#include "EventEmitter.h"
+#include "JSDOMWrapperCache.h"
+#include "JSEventListener.h"
+
+namespace WebCore {
+using namespace JSC;
+
+JSValue toJSNewlyCreated(JSGlobalObject*, JSDOMGlobalObject* globalObject, Ref<EventEmitter>&& value)
+{
+    return createWrapper<EventEmitter>(globalObject, WTFMove(value));
+}
+
+EventEmitter* JSEventEmitter::toWrapped(VM& vm, JSValue value)
+{
+    if (value.inherits<JSEventEmitter>())
+        return &jsCast<JSEventEmitter*>(asObject(value))->wrapped();
+    return nullptr;
+}
+
+std::unique_ptr<JSEventEmitterWrapper> jsEventEmitterCast(VM& vm, JSC::JSGlobalObject* lexicalGlobalObject, JSValue thisValue)
+{
+    if (auto* target = jsDynamicCast<JSEventEmitter*>(thisValue))
+        return makeUnique<JSEventEmitterWrapper>(target->wrapped(), *target);
+    if (auto* object = jsDynamicCast<JSObject*>(thisValue)) {
```

Comment:
```suggestion
    if (auto* object = jsDynamicCast<JSNonFinalObject*>(thisValue)) {
```

Reply:
I think that will handle the case where it's a constructor or prototype

Reply:
fixed.